### PR TITLE
feat(next): make astro:env stable

### PR DESCRIPTION
**Pull request by florian-lefebvre**
## Changes

- Makes `astro:env` stable
- Virtual module and types are now always present by default
- Manual chunk for `astro/env/setup` is removed, alongside TLA in adapters. Peer deps are bumped for adapters
- Depends on #11771

## Testing

Tests should still pass, manually tested

## Docs

Removes a lot of JSDoc since it will move to a proper docs page

### CODE REVIEW

File: packages/integrations/vercel/src/serverless/entrypoint.ts [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
 	ASTRO_PATH_PARAM,
 } from './adapter.js';
 
-// Won't throw if the virtual module is not available because it's not supported in
-// the users's astro version or if astro:env is not enabled in the project
-await import('astro/env/setup')
-	.then((mod) => mod.setGetEnv((key) => process.env[key]))
-	.catch(() => {});
-
+setGetEnv((key) => process.env[key]);
```
Comment (florian-lefebvre) [inline packages/integrations/vercel/src/serverless/entrypoint.ts:12]: Should it run after the polyfill?

File: packages/astro/src/core/build/plugins/plugin-chunks.ts [hunk 2]
```diff
@@ -16,11 +16,6 @@ export function vitePluginChunks(): VitePlugin {
 					if (id.includes('astro/dist/runtime')) {
 						return 'astro';
 					}
-					// Place `astro/env/setup` import in its own chunk to prevent Rollup's TLA bug
-					// https://github.com/rollup/rollup/issues/4708
-					if (id.includes('astro/dist/env/setup')) {
```
[the remark this answers is not in the document]
Comment (bluwy) [reply packages/astro/src/core/build/plugins/plugin-chunks.ts:21]: If we require that every adapter will now need to support astro env and statically import it, then it should be safe 👍

File: packages/integrations/vercel/src/serverless/entrypoint.ts (hunk 1 above)
Comment (bluwy) [reply packages/integrations/vercel/src/serverless/entrypoint.ts:12]: Yeah it probably should

File: packages/astro/src/core/build/plugins/plugin-chunks.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (florian-lefebvre) [reply packages/astro/src/core/build/plugins/plugin-chunks.ts:21]: okay perfect! the only remaining one will be netlify

File: packages/astro/src/env/sync.ts [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
-		return;
-	}
-
-	const schema = settings.config.experimental.env.schema ?? {};
-
-	let client = '';
-	let server = '';
+export function syncAstroEnv(settings: AstroSettings): void {
+	let client: string | null = null;
+	let server: string | null = null;
```
Comment (bluwy) [inline packages/astro/src/env/sync.ts:7]: Is there a reason to do `string | null = null` instead of `string = ''`? It seems to make the code below more complex.

File: .changeset/eighty-boxes-applaud.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+export default defineConfig({
+-    experimental: {
+        env: {
+            schema: {
+                FOO: envField.string({ /* ... */ })
+            }
+        }
+-    }
+})
```
Comment (bluwy) [inline .changeset/eighty-boxes-applaud.md:20]: small nit: i think we usually use two spaces (tab size) for the indent

**Review by bluwy — APPROVED** [review]
LGTM. We can merge once the call for consensus ends and the RFC is merged.

File: .changeset/eighty-boxes-applaud.md
Comment (ematipico) [inline .changeset/eighty-boxes-applaud.md:1]: The changeset should follow this template: https://contribute.docs.astro.build/docs-for-code-changes/experimental-feature-docs/#changeset

File: packages/integrations/node/package.json [hunk 5]
```diff
@@ -37,7 +37,7 @@
     "server-destroy": "^1.0.1"
   },
   "peerDependencies": {
-    "astro": "^4.2.0"
+    "astro": "^5.0.0-alpha.0"
```
Comment (ematipico) [inline packages/integrations/node/package.json:40]: We'll have to remember to update this once we go to beta

File: .changeset/eighty-boxes-applaud.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+---
+
+<script>
+import { API_URL } from "astro:env/client"
+
+fetch(`${API_URL}/ping`)
+</script>
+```
+
+If you were previously using this feature, please remove the experimental flag from your Astro config:
```
Comment (florian-lefebvre) [inline .changeset/eighty-boxes-applaud.md:30]: How should I rephrase this to say it should just be moved out of the experimental object, not removed?

File: packages/astro/src/types/public/config.ts [hunk 7]
```diff
@@ -1390,6 +1390,76 @@ export interface AstroUserConfig {
 	 */
 	legacy?: object;
 
+	/**
```
Comment (sarah11918) [inline packages/astro/src/types/public/config.ts:1393]: Just checking on this position, because it seems like this would make it show up underneath "Legacy Flags". Did you mean to put it here, or after i18n?

(Can also be put in any position in the main list, as you think it seems appropriate.)

File: packages/astro/src/types/public/config.ts [hunk 8]
```diff
@@ -1390,6 +1390,76 @@ export interface AstroUserConfig {
...
+		 * @version 5.0.0
+		 * @description
+		 *
+		 * An object that uses `envField` to define the data type (`string`, `number`, or `boolean`) and properties of your environment variables: `context` (client or server), `access` (public or secret), a `default` value to use, and whether or not this environment variable is `optional` (defaults to `false`).
+		 * ```js
+		 * // astro.config.mjs
+		 * import { defineConfig, envField } from "astro/config"
+		 *
+		 * export default defineConfig({
+		 *   experimental: {
```
Comment (sarah11918) [inline packages/astro/src/types/public/config.ts:1418]: Make sure to update the code samples to no longer show this under `experimental`!

File: packages/astro/src/types/public/config.ts [hunk 9]
```diff
@@ -1390,6 +1390,76 @@ export interface AstroUserConfig {
...
+		 * Whether or not to validate secrets on the server when starting the dev server or running a build.
+		 *
+		 * By default, only public variables are validated on the server when starting the dev server or a build, and private variables are validated at runtime only. If enabled, private variables will also be checked on start. This is useful in some continuous integration (CI) pipelines to make sure all your secrets are correctly set before deploying.
+		 *
+		 * ```js
+		 * // astro.config.mjs
+		 * import { defineConfig, envField } from "astro/config"
+		 *
+		 * export default defineConfig({
+		 *   experimental: {
```
Comment (sarah11918) [inline packages/astro/src/types/public/config.ts:1449]: experimental!

**Review by sarah11918 — COMMENTED** [review]
Thanks @florian-lefebvre !

Made some comments below to get this formatted just right. 

Also double checking there are no error messages associated with `env`? I don't see any here, but if there are, they'd probably need updating too (they might point to the experimental link etc.)

File: packages/astro/src/core/errors/errors-data.ts
Comment (florian-lefebvre) [inline packages/astro/src/core/errors/errors-data.ts:1]: @sarah11918 some errors there indeed!

File: .changeset/eighty-boxes-applaud.md (hunk 6 above)
Comment (sarah11918) [reply .changeset/eighty-boxes-applaud.md:30]: There's precedent with i18n coming out of experimental! I'll hunt that down!

**Review by sarah11918 — COMMENTED** [review]
@florian-lefebvre just pushed a commit to reorder some of the stuff in the changeset. See what you think, and we can edit as you like!

**Comment by florian-lefebvre** [comment]
LGTM! If this looks good to you, I'll merge tomorrow

File: .changeset/eighty-boxes-applaud.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    schema: {
+      API_URL: envField.string({ context: "client", access: "public", optional: true }),
+      PORT: envField.number({ context: "server", access: "public", default: 4321 }),
+      API_SECRET: envField.string({ context: "server", access: "secret" }),
+    }
+  }
+})
+```
+
+Then, you can import and use your defined variables from the appropriate `/client` or `/server` module:
```
Comment (sarah11918) [inline .changeset/eighty-boxes-applaud.md:25]:
```suggestion
You can import and use your defined variables from the appropriate `/client` or `/server` module:
```

**Review by sarah11918 — APPROVED** [review]
Approving for docs!
